### DISCUSSION TITLE
understanding Undercollateralization, Minting, Burning, Liquidation of DSC/ Foundry-Defi-StableCoin

### DISCUSSION THREAD

**Discussion by komalrao1** [Cyfrin/foundry-full-course-cu#2531] [Q&A] [answered]
I'm facing difficulty understanding how the contract works when our deposits become undercollateralized. What I have understood is that when our deposits become undercollateralized, someone can liquidate our DSC tokens and, in return, they will receive all of our collateral, and we won't get any deposit back. The liquidation part is okay, but what I don't understand is how this is going to be a stable coin. Because when someone deposits some collateral and gets some DSC tokens, and later if the value of the collateral decreases, they either need to add more collateral or risk liquidation.

In the above case, the stable coin is directly dependent on the prices of collaterals. If the price changes, then the DSC tokens' positions are also changing. Why do we call this a stable coin?

**Comment by EngrPips** [comment]
Honestly, this is more about you not understanding what it means for something to be stable than an issue with the course codebase. We call the DSC a stablecoin because, at every point in time, we always have enough collateral reserve so that a single `DSC` is always equivalent to `1$`. I mean, you can conclude you have a stable network if, at every point in time, every network operation you carry out on your device is smooth and successful. So, for anything to be stable, it must meet the criteria that we have concluded as stability.

**Reply by komalrao1** [reply]
I didn't mean for an issue it the code base i should have asked this in general . I couldn't properly understand the proper purpose of stable coin .  We are depositing some Weth and minting some **DSC** coins. If the value of deposited Weth is decreasing either we need  to increase deposit or some one will liquidate the DSC coins. Here **DSC** coins value depended on Weth value. If it is depended on Weth then why are we using stable coins what is the purpose of it?

**Reply by EngrPips** [reply, marked as answer]
Honestly, I believe Patrick did a great job explaining what stablecoin is and why we need it in the course, but maybe you missed the explanation he did. I will try to explain as much as I can. Primarily, what I use stablecoin for is to reserve my money. I mean, If I have $1000 and I want to keep it safe without the intention of investing, then I will go buy some stablecoin, and by doing that, I can be assured that my $1000 is always $1000. Talking about stablecoins, they need to be backed by something, so some stablecoins are backed by dollar notes in some bank accounts while some are backed by other cryptocurrencies, and some are even backed by some algorithm that adjusts the quantity of the stablecoin based on the demand and supply. This Stable coin protocol that we built in this course doesn't seem to be something that we use to reserve our money as it is directly dependent on the collateral we deposit into the protocol, but it seems to make more sense in terms of farming opportunity in the `Defi` space. If some protocol was allowing you to stake `DSC` to earn some profit, then you would be incentivized to mint `DSC` and deposit your collateral to make a profit with your initial capital.

**Reply by komalrao1** [reply]
Got it . Thank you for the explanation. I have gone through video explained by patrick three times. I fell like i understood how a stable coin should work based on example provided by patrick related to apple coin. I was confused because of dependency of DSC coin on Weth and Wbtc. What i understood is when we buy some stable coin it should have same buying power even after many years. 

Can you suggest any more resources related to stable coin?

**Reply by EngrPips** [reply]
Below are links to some content that can help you dive deep into what stablecoins are.

https://www.simplilearn.com/tutorials/blockchain-tutorial/what-is-a-stablecoin https://www.youtube.com/watch?v=LOgD9ZexxC4 https://www.youtube.com/watch?v=GsSSLDzKCOE
